test/jb_controller/change_token_of.test.js

```diff
@@ -0,0 +1,238 @@
+import { packFundingCycleMetadata } from '../helpers/utils';
+import errors from '../helpers/errors.json';
+
+import jbDirectory from '../../artifacts/contracts/JBDirectory.sol/JBDirectory.json';
+import jbFundingCycleStore from '../../artifacts/contracts/JBFundingCycleStore.sol/JBFundingCycleStore.json';
+import jbOperatoreStore from '../../artifacts/contracts/JBOperatorStore.sol/JBOperatorStore.json';
+import jbProjects from '../../artifacts/contracts/JBProjects.sol/JBProjects.json';
+import jbSplitsStore from '../../artifacts/contracts/JBSplitsStore.sol/JBSplitsStore.json';
+import jbToken from '../../artifacts/contracts/JBToken.sol/JBToken.json';
+import jbTokenStore from '../../artifacts/contracts/JBTokenStore.sol/JBTokenStore.json';
+
+describe('JBController::changeTokenOf(...)', function () {
+  const PROJECT_ID = 1;
+  const DOMAIN = 1;
+  const NAME = 'TestTokenDAO';
+  const SYMBOL = 'TEST';
+  let CHANGE_TOKEN_INDEX;
+
+  before(async function () {
+    let jbOperationsFactory = await ethers.getContractFactory('JBOperations');
+    let jbOperations = await jbOperationsFactory.deploy();
+
+    CHANGE_TOKEN_INDEX = await jbOperations.CHANGE_TOKEN();
+  });
+
+  async function setup() {
+    let [deployer, projectOwner, ...addrs] = await ethers.getSigners();
+
+    const blockNum = await ethers.provider.getBlockNumber();
+    const block = await ethers.provider.getBlock(blockNum);
+    const timestamp = block.timestamp;
+
+    let [
+      mockJbDirectory,
+      mockJbFundingCycleStore,
+      mockJbOperatorStore,
+      mockJbProjects,
+      mockJbSplitsStore,
+      mockJbToken,
+      mockJbTokenStore,
+    ] = await Promise.all([
+      deployMockContract(deployer, jbDirectory.abi),
+      deployMockContract(deployer, jbFundingCycleStore.abi),
+      deployMockContract(deployer, jbOperatoreStore.abi),
+      deployMockContract(deployer, jbProjects.abi),
+      deployMockContract(deployer, jbSplitsStore.abi),
+      deployMockContract(deployer, jbToken.abi),
+      deployMockContract(deployer, jbTokenStore.abi),
+    ]);
+
+    let jbControllerFactory = await ethers.getContractFactory('JBController');
+    let jbController = await jbControllerFactory.deploy(
+      mockJbOperatorStore.address,
+      mockJbProjects.address,
+      mockJbDirectory.address,
+      mockJbFundingCycleStore.address,
+      mockJbTokenStore.address,
+      mockJbSplitsStore.address,
+    );
+
+    await mockJbTokenStore.mock.issueFor
+      .withArgs(PROJECT_ID, NAME, SYMBOL)
+      .returns(mockJbToken.address);
+
+    await mockJbProjects.mock.ownerOf.withArgs(PROJECT_ID).returns(projectOwner.address);
+
+    return {
+      projectOwner,
+      addrs,
+      jbController,
+      mockJbOperatorStore,
+      mockJbFundingCycleStore,
+      mockJbTokenStore,
+      mockJbToken,
+      timestamp,
+    };
+  }
+
+  it(`Should change current token if caller is project owner and funding cycle not paused`, async function () {
```

Comment:
"not paused" state shouldn't matter here.

Reply:
ahh, this is referring to it being allowed or not via "changeTokenAllowed" flag. got it.